src/index.spec.js

```diff
@@ -170,7 +177,39 @@ var serviceWorkerOption = {
             )
           }
         )
-      })
+      }),
+        it('should get passed assetsHash', done => {
+          const transformOptions = serviceWorkerOption => {
+            expect(serviceWorkerOption)
+              .to.have.property('assetsHash')
+              .that.is.a('string')
+          }
+
+          const serviceWorkerPlugin = new ServiceWorkerPlugin({
+            filename,
+            transformOptions,
+          })
+
+          const compilation = {
+            options: fakeOptions,
+            assets: {
+              [filename]: {
+                source: () => '',
```

Comment:
Perhaps you can add a second test case, with a different mocked source, as evidence that the hash is something else when the contents change.

Reply:
Oh I think we've a design decision here. Maybe I didn't explain fully, but currently only the filenames get hashed not the contents. In my project I use filenames with hashes so there is no problem with this. On the other hand after reading [this](https://jakearchibald.com/2016/caching-best-practices/#a-service-worker-can-extend-the-life-of-these-bugs) I think the only real use case to make the assetsHash include the content would be to make the serviceworker byte different when the assets change, but not their filenames. I nthis case it would probably be easier to just make the `jsonStats` include the compilation hash by changing [this line](https://github.com/oliviertassinari/serviceworker-webpack-plugin/blob/master/src/index.js#L144) to true

Reply:
Think I indeed misunderstood the point of your PR, but yes - my approach would be to have some kind of value, that's the same between builds of the same code, but different when sourcecode changes.

Tbh; I'm not that familiar with webpack internals, but it sounds like enabling the hash would solve the same problem as your PR does. 

@devCrossNet @oliviertassinari What do you say?